src/options.cpp

```diff
@@ -2516,6 +2527,15 @@ std::string options_manager::show( bool ingame, const bool world_options_only )
 
 #if !defined(__ANDROID__) && (defined TILES || defined _WIN32 || defined WINDOWS)
     if( terminal_size_changed ) {
+        int scaling_factor = get_scaling_factor();
+        int TERMX = ::get_option<int>( "TERMINAL_X" );
+        int TERMY = ::get_option<int>( "TERMINAL_Y" );
+        TERMX -= TERMX % scaling_factor;
+        TERMY -= TERMY % scaling_factor;
+        get_option( "TERMINAL_X" ).setValue( std::max( 80 * scaling_factor, TERMX ) );
+        get_option( "TERMINAL_Y" ).setValue( std::max( 24 * scaling_factor, TERMY ) );
```

Comment:
You can replace `80` and `24` magic numbers with `FULL_SCREEN_WIDTH` and `FULL_SCREEN_HEIGHT` respectively.